project_task_code/tests/test_project_task_code.py

```diff
@@ -34,3 +35,19 @@ def test_copy_task_code_assign(self):
         project_task_copy = self.project_task.copy()
         self.assertNotEqual(project_task_copy.code, self.project_task.code)
         self.assertEqual(project_task_copy.code, code)
+
+    def test_name_search(self):
+        code_list = []
+        tasks = self.project_task_model
+        all_tasks = tasks.search([])
+        for i in range(10):
+            task = tasks.create({'name': 'task0%s' % str(i + 1)})
+            code_list.append(task.code)
+            tasks += task
+        task_names = [task.name for task in tasks]
+        for code in code_list:
+            [(task_id, task_name)] = tasks.name_search(name=code)
+            self.assertIn(task_id, tasks.ids)
+            self.assertNotIn(task_id, all_tasks.ids)
+            self.assertIn(task_name, task_names)
+        self.assertEqual(tasks + all_tasks, tasks.search([]))
```

Comment:
these tests are not very interesting. you should indeed test that it works to find a task by test, but if this woks one time or 10 times doesn't matter so much.

It matters much more that the test tests that searching for something not like a code yields the right products still (and only those).